src/colinajose/model/notification/Device.java

```diff
@@ -0,0 +1,57 @@
+package colinajose.model.notification;
+
+import colinajose.model.people.Contact;
+import colinajose.model.people.Person;
+import colinajose.model.people.Student;
+import datastructures.circulardoublylinkedlist.MyCircularDoublyLinkedList;
+
+public class Device implements Observer{
+    private Observable studentsState;
+    private Person owner;
+
+    public Device(Observable studentsState, Person owner) {
+        this.studentsState = studentsState;
+        this.owner = owner;
+    }
+
+    @Override
+    public void update() {
+        String state = (String) this.studentsState.getState(owner);
+        if(state.equals("NOTIFY")){
+            sendNotification();
+        }
+    }
+
+    public void sendNotification(){
+        System.out.println("Sending notification for student: " + owner.getName());
+        MyCircularDoublyLinkedList parents = ((Student)owner).getParents();
```

Comment:
owner should be a Parent or a Student not specifically an student. When you are registering an student you should register its parent the devices that each of them have. This class has nothing to do with Parent or student we only care that owner is a Person